src/Fields/Media.php

```diff
@@ -139,6 +146,52 @@ private function setOrder($ids)
         $mediaClass::setNewOrder($ids);
     }
 
+    private function addExistingMedia(NovaRequest $request, $data, HasMedia $model, string $collection, Collection $medias): Collection
+    {
+        $added_media_ids = $medias->pluck('id')->toArray();
+
+        return collect($data)
+            ->filter(function ($value) use ($added_media_ids) {
+                return (!($value instanceof UploadedFile)) && !(in_array((int) $value, $added_media_ids));
+            })->map(function ($model_id, int $index) use ($request, $model, $collection) {
+                $mediaClass = config('medialibrary.media_model');
+                $existing_media = $mediaClass::find((int) $model_id);
```

Comment:
I just put some code on master so UUID o the Media would work. Here, we need to get rid of the integer casting.